Pull Request - Edwin Branch

edwinperaza opened the pull request:
Please @josemontesdeoca check this program.

edwinperaza committed:
Initial Commit

edwinperaza committed:
Updated gitignore

edwinperaza committed:
New Search Client created

edwinperaza committed:
scroll down “infinitely” to continue loading more image results (up to 8 pages)

edwinperaza committed:
User can click on settings which allows selection of advanced search options to filter results

edwinperaza committed:
User can configure advanced search filters

edwinperaza committed:
Check if internet is available, handle error cases

edwinperaza committed:
Last commit

josemontesdeoca commented on README.md:
```diff
@@ -2,22 +2,23 @@
+ * [X] Advanced: Robust error handling, check if internet is available, handle error cases, network failures
  * [ ] Advanced: Use the ActionBar SearchView or custom layout as the query box instead of an EditText
  * [ ] Advanced: User can share an image to their friends or email it to themselves
  * [ ] Advanced: Replace Filter Settings Activity with a lightweight modal overlay
```
@edwinperaza you should mark this as you did the filter with a `DialogFragment` instead of a separate activity

josemontesdeoca commented on app/build.gradle:
```diff
@@ -0,0 +1,27 @@
+
+    defaultConfig {
+        applicationId "com.example.edwinmperazaduran.gridimagesearch"
+        minSdkVersion 14
```
@edwinperaza Nowadays you should target 16 and above. 14 is pretty old and not worth the extra work to support it

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java:
```diff
@@ -0,0 +1,155 @@
+
+public class ImageFilterDialog extends DialogFragment{
+
+    private Spinner sp_imgsize;
```
@edwinperaza follow Google's recommended parameter name styleguide https://google.github.io/styleguide/javaguide.html#s5.2.6-parameter-names

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java:
```diff
@@ -0,0 +1,155 @@
+        if (getArguments().getSerializable("imageFilter") != null) {
+            imageFilter = (ImageFilter) getArguments().getSerializable("imageFilter");
+        }
+        //hideSoftKeyboard(view);
```
@edwinperaza why this commented code?

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java:
```diff
@@ -0,0 +1,155 @@
+
+    public void setupSpAdapter(View view){
+
+        ArrayAdapter<CharSequence> adaptercolor = ArrayAdapter.createFromResource(view.getContext(),
```
@edwinperaza as your spinners options are static. you are better off defining them within the layout. I'll add a comment on the layout on how you can do that. You do not need to programmatically create this Adapters here.

josemontesdeoca commented on app/src/main/res/layout/fragment_image_filter.xml:
```diff
@@ -0,0 +1,125 @@
+        android:textSize="18sp"
+        android:minHeight="45dp"
+        android:paddingBottom="10dp"
+        android:layout_toRightOf="@+id/tv_imgcolor" />
```
@edwinperaza you can initialize the spinners with static options via the `android:entries=` property. Fro example, in this particular spinner you can do `android:entries=@array/sp_img_sz`. Similar treatment can be done to the other spinners.

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java:
```diff
@@ -0,0 +1,155 @@
+        sp_img_type.setAdapter(adaptertype);
+
+        if (imageFilter != null){
+            for (int i = 0; i < adaptercolor.getCount(); i++) {
```
@edwinperaza this is pretty inefficient. The adapter already has a method to get the position of an item, you can use that to get the position and ten set the selection. ie:

``` java
int post = adaptercolor.getPosition(imageFilter.getColor());
sp_imgcolor.setSelection(pos)
```

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java:
```diff
@@ -0,0 +1,155 @@
+        dismiss();
+    }
+
+    public void setupSpAdapter(View view){
```
@edwinperaza the method name is pretty ambiguous. you should err on the side of well written method names plus simple [Javadoc](https://google.github.io/styleguide/javaguide.html#s7-javadoc) to understand what the method does.

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java:
```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    public static void hideSoftKeyboard(View view) {
```
@edwinperaza you may want to have this on a separate Helper utility class as this may be used often in other places.

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/ImageDisplayActivity.java:
```diff
@@ -0,0 +1,50 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_image_display);
+        //String url = getIntent().getStringExtra("url");
```
@edwinperaza comment?

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java:
```diff
@@ -0,0 +1,179 @@
+        ImageFilterDialog.ImageFilterDialogListener{
+
+    private static int MAX_PAGE = 8;
+    EditText etQuery;
```
@edwinperaza you should be explicit on the `public` or `private` matter of the variable.

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java:
```diff
@@ -0,0 +1,179 @@
+            @Override
+            public void onLoadMore(int page, int totalItemsCount) {
+                if (page < MAX_PAGE) {
+                    onImageSearch(page - 1);
```
@edwinperaza -1 logic should be handled inside the `onImageSearch()` method. This way you just pass in the page and thats it.

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java:
```diff
@@ -0,0 +1,179 @@
+
+                            @Override
+                            public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                                super.onFailure(statusCode, headers, responseString, throwable);
```
@edwinperaza you should handle the `onFailure` of the API call and give the user some visual queue of what happened

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java:
```diff
@@ -0,0 +1,179 @@
+    public static void hideSoftKeyboard(View view) {
+        InputMethodManager inputMethodManager = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+        inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
+
```
@edwinperaza remove extra lines here

josemontesdeoca commented on app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java:
```diff
@@ -0,0 +1,179 @@
+    }
+
+    public void onFinishDialog(ImageFilter objFilter){
+
```
@edwinperaza remove extra line)

josemontesdeoca commented:
@edwinperaza initial review is done. Let me know if you have any questions from the comments added. Look forward your fixes to take another look

edwinperaza committed:
Update target to 16

edwinperaza committed:
Update: Using Google Custom Search

edwinperaza committed:
Update: User can configure advanced search filters with new API and ScrollListener

edwinperaza committed:
Updating README.md

edwinperaza committed:
Adding data to spinner more efficient
Update Dialog Fragment to show TextView
Update ImageDisplayActivity to increase the image's size
Adding more string resource

edwinperaza committed:
Add message to onFailure in onImageSearch method
